test/benchmark/avro-upsert/metric-verifier/wait_metric

```diff
@@ -10,32 +10,63 @@
 # by the Apache License, Version 2.0.
 
 import argparse
+import dateparser
+import datetime
 import sys
 import time
+import urllib
 
 import prometheus_api_client
+import prometheus_api_client.utils
 
 def run(args: argparse.Namespace) -> int:
-    """Run the generator, inserting args.num_records number of messages."""
+    """Wait for the query to settle or timeout and then dump ingest metrics."""
 
-    prom = prometheus_api_client.PrometheusConnect(f'http://{args.host}:{args.port}')
+    try:
+        start = dateparser.parse('30 seconds ago')
+        return wait_metric(args)
+    finally:
+        print_dashboard_url(args, start)
+
+
+def print_dashboard_url(args: argparse.Namespace, start: datetime.datetime):
+    """Print the URL to see the metrics for this test"""
 
+    # Create parameters to see a dashboard with the metrics from this benchmark run
+    # Convert start and end to timestamps, with a little padding to make the dashboard look nicer
+    # Grafana expects timestamps with milliseconds
+    path = '/d/materialize-overview/materialize-overview'
+    query = urllib.parse.urlencode( {
+              "from": round(start.timestamp()) * 1000,
+                "to": round(dateparser.parse('in 30 seconds').timestamp()) * 1000,
+                "tz": "UTC"
+             })
+    dashboard_url = urllib.parse.urlunparse(('http', args.grafana_location, path, '', query, ''))
+    print(f'View performance metrics here: {dashboard_url}')
+
+
+def wait_metric(args: argparse.Namespace) -> int:
+    """Wait for the given metric, returning desired exit code (0 is success)."""
+    prom = prometheus_api_client.PrometheusConnect(f'http://{args.host}:{args.port}')
     time_so_far = 0
     start = int(time.monotonic())
     while time_so_far < args.timeout_seconds:
         current_values = prom.custom_query(args.prom_query)
         if args.verbose:
             print(current_values)
 
-        if len(current_values) != 1:
-            print('ERROR: Prometheus query must only return a single result!')
+        if len(current_values) > 1:
+            print('ERROR: Prometheus query must only return a zero or one results!')
+            prometheus_api_client.utils.pretty_print_metric(current_values)
             return 1
 
         # We aren't running query_range, so there should only be a single timestamp and point in the reponse
-        (ts, point) = current_values[0]['value']
-        if point == args.expected_value:
-            print(f"SUCCESS! Query settled in {time_so_far} seconds")
-            return 0
+        if current_values:
+            (ts, point) = [float(i) for i in current_values[0]['value']]
+            if point == args.expected_value:
+                rate = point / max(time_so_far, 1)
+                print(f"SUCCESS! Benchmark took {time_so_far}s, {rate} rows/sec")
```

Comment:
minor nit: would love for the rows / second to be rounded to the nearest row

Reply:
Sure thing!